Added Dark Mode Support

Tom-Mouse-p opened the pull request:
# Add Dark Mode Support with Next Themes
Closes #7

## Changes

- Introduced dark mode support using `next-themes` library
- Added new dependency: `next-themes`

## New Components

- `components/theme-provider.tsx`: Provider component to manage theme state
- `components/ui/dropdown-menu`: ShadCn UI component
- `components/theme-toggle.tsx`: Toggle button for switching between light and dark modes

## Benefits

This PR enhances the user experience by allowing users to switch between light and dark themes based on their preference. The implementation follows best practices for theme management in Next.js applications.

Tom-Mouse-p committed:
Added Dark Mode Support

aramshiva committed:
Merge branch 'main' into feature/darkmode

Tom-Mouse-p committed:
Merge remote-tracking branch 'upstream/main' into feature/darkmode

Tom-Mouse-p committed:
Merge branch 'feature/darkmode' of https://github.com/Tom-Mouse-p/contribution-nomen into feature/darkmode

aramshiva commented:
thanks for the PR -- looks like the PR is failing from a separate error on my part. I'll fix it and get back to you.

aramshiva committed:
Remove Geo

aramshiva reviewed (changes requested):
Thanks for your PR! I am getting the following from the header and the top in dark mode where it's white:
![image](https://github.com/user-attachments/assets/d556df6d-47ee-4c54-8c1a-5a23fdba76c7)
Once that's fixed, I can merge :)

Tom-Mouse-p committed:
Theme fix and Add TopBar component and integrate it into the Popular Names page

Tom-Mouse-p committed:
Merge remote-tracking branch 'upstream/main' into feature/darkmode

Tom-Mouse-p commented:
# Add Dark Mode Support with Next Themes
Closes #7

## Changes

- Introduced dark mode support using `next-themes` library
- Created a reusable TopBar component for consistent navigation
- Added theme variables to support both light and dark modes
- Updated styling across pages to use theme CSS variables instead of hardcoded colors
- Added system theme detection alongside manual theme selection

## New Components

- `components/top-bar.tsx`: Reusable navigation bar with integrated theme toggle
- `components/theme-toggle.tsx`: Toggle button for switching between light, dark, and system modes

## Benefits

This PR enhances the user experience by allowing users to switch between light, dark, and system themes based on their preference. The implementation maintains all existing functionality while providing a more comfortable viewing experience in different lighting conditions.

aramshiva reviewed (changes requested):
It's erroring with:
```
Failed to compile.
./app/page.tsx
26:10  Error: 'ThemeToggle' is defined but never used.  @typescript-eslint/no-unused-vars
81:6  Warning: React Hook useEffect has a missing dependency: 'handleSearch'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
./app/popular/page.tsx
21:10  Error: 'motion' is defined but never used.  @typescript-eslint/no-unused-vars
23:10  Error: 'ThemeToggle' is defined but never used.  @typescript-eslint/no-unused-vars
101:6  Warning: React Hook useEffect has a missing dependency: 'fetchPopularNames'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
105:6  Warning: React Hook useEffect has a missing dependency: 'fetchPopularNames'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
Error: Command "npm run build" exited with 1
```

Tom-Mouse-p committed:
Solves Build Errors & Warnings - Add dynamic export to API routes and clean up query conditions

Tom-Mouse-p commented:
Does that work now?

# Solved Build Errors & Warnings
1. Solved build error - Removed unused imports
2. Solved Build Warnings - added `export const dynamic = "force-dynamic";` to the api routes

aramshiva commented on app/layout.tsx:
```diff
@@ -28,11 +29,13 @@ export default function RootLayout({
         className={`${geistSans.variable} ${geistMono.variable} antialiased`}
       >
-        <AnimatePresence>{children}</AnimatePresence>
+        <ThemeProvider attribute="class" defaultTheme="system" enableSystem>
```
```suggestion
        <ThemeProvider attribute="class" defaultTheme="light" enableSystem>
```

aramshiva committed:
Update app/layout.tsx

aramshiva reviewed (approved):
Thank you for your contribution!
